tools/clang_format.rst

```diff
@@ -5,22 +5,38 @@ Using clang-format for LSST Development
 #######################################
 
 The easiest way to comply with the :ref:`LSST code layout rules <style-guide-cpp-6-1>` is to use `clang-format <http://clang.llvm.org/docs/ClangFormat.html>`_.
+We require clang-format version 5.0, for formatting consistency (earlier versions may format some lines subtly differently).
 
-You can install this tool through the Ubuntu package manager:
+Ubuntu users can install this tool through the Ubuntu package manager.
+Version 5.0 is not available in the default Xenial or Zesty repositories.
+You can get it from the `llvm apt repository <http://apt.llvm.org/>`_, using the 5.0 repository (qualification branch).
+Once you've added the appropriate apt source line for your Ubuntu version, install it using either the Xenial (16.04):
 
 .. code-block:: bash
 
-    sudo apt-get install clang-format
+    echo "deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-5.0 main" | sudo tee /etc/apt/sources.list.d/llvm.list
+    wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
+    sudo apt-get update
+    sudo apt-get install clang-format-5.0
```

Comment:
re: `apt-get`, force of habit from many years of Debian (I had to look up the difference between them just now). I changed it to `apt`.

Yes, this will install `clang-format-5.0` as a binary, with no symlinks for `clang-format`. If you have no other `clang-format`s available, you should just be able to tab-complete it as normal. I think this is llvm's way of managing the various versions they provide on their repository.

I've added a note about using `update-alternatives` to add a system symlink. What do you think?

Reply:
Wow...

    /usr/bin/clang-format -> /etc/alternatives/clang-format
    /etc/alternatives/clang-format -> /usr/bin/clang-format-5.0

Is that really the preferred way to do things?

(And yes, it does work.)

Reply:
Yes, nested symlinks like that are how `update-alternatives` works (it's been a Debian standard tool for years). It then only has to manage the `/etc/alternatives` link, and never touch the `/usr/bin` link, once it's established.